src/Microsoft.AspNetCore.SignalR.Core/DefaultHubLifetimeManager.cs

```diff
@@ -123,17 +113,20 @@ public override Task InvokeGroupAsync(string groupName, string methodName, objec
                 throw new ArgumentNullException(nameof(groupName));
             }
 
-            return InvokeAllWhere(methodName, args, connection =>
+            List<Task> tasks = new List<Task>();
+            InvocationMessage message = CreateInvocationMessage(methodName, args);
+            var group = _groups[groupName];
+            if (group != null)
             {
-                var feature = connection.Features.Get<IHubGroupsFeature>();
-                var groups = feature.Groups;
+                tasks = group.Values.Select(c => WriteAsync(c, message)).ToList();
+            }
 
-                // PERF: ...
-                lock (groups)
-                {
-                    return groups.Contains(groupName) == true;
-                }
-            });
+            return Task.WhenAll(tasks);
```

Comment:
Move this inside the `if (group != null)` and after the `if` check have a `return Task.CompletedTask` for cases when there is no group.